app/services/draw_activator.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+#
+# Service object to handle the activation of a draw. Checks to make sure the
+# draw has the correct status, updates the status, and sends e-mail invitations
+# to students in the draw.
+class DrawActivator
+  # Class method to permit calling :activate on the class without instantiating
+  # the service object directly
+  #
+  # @attr draw [Draw] the draw in question
+  def self.activate(draw:)
+    new(draw: draw).activate
+  end
+
+  # Initialize a new DrawActivator
+  #
+  # @attr draw [Draw] the draw in question
+  # @attr mailer [ActionMailer::Base] mailer class for sending invitation
+  #   e-mails
+  def initialize(draw:, mailer: StudentMailer)
+    @draw = draw
+    @errors = []
+    @mailer = mailer
+  end
+
+  # Activate a Draw
+  #
+  # @return [Hash{Symbol=>ApplicationRecord,Hash}] A results hash with the
+  #   message to set in the flash and either `nil` or the created object.
+  def activate
+    validate
+    activate_draw if valid?
+    send_emails if valid?
+    valid? ? success : error
```

Comment:
~~I think it makes sense to wrap this all in one `if`~~

hmm, this isn't the biggest problem, but it's unclear that validity can change between the steps of this method. Not quite sure what to suggest, and it's also fine as it is -- just something I thought while reading through the class